### PR TITLE
[JENKINS-63766] Work around JDK-8231454

### PR DESCRIPTION
**Pull request by basil**
See https://github.com/jenkinsci/workflow-cps-plugin/pull/543. The short summary is that until JDK-8231454 gets backported to `jdk11u-dev`, Jenkins users running Pipeline jobs in Java 11 (but not Java 8 or Java 17) are vulnerable to a metaspace memory leak. https://github.com/jenkinsci/workflow-cps-plugin/pull/543 works around the problem for Java 11 users via reflection, but that in turns requires an addition to our `--add-opens` directives, which is being done in this PR. I plan to backport this to LTS.

### Proposed changelog entries

Provide supporting infrastructure to enable Pipeline: Groovy to work around a metaspace memory leak (JDK-8231454) for users running Pipeline jobs on Java 11.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

### CODE REVIEW

**Review by jglick — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
> means it has been accepted, so we could choose to remove the workaround (here and in plugins) once that update has been released, right?

and given some time for adoption yes.
Can take awhile for different packages to appear throughout distributions and OS variants.

**Comment by basil** [comment]
My understanding (only 85% sure about this) is that `jdk11u-dev` is currently tracking 11.0.17 with a projected October 18. 2022 GA. So we'll have to wait a while before the upstream fix propagates to Linux distributions and Docker images.

**Comment by jglick** [comment]
Oh, so quite a ways out actually.

**Pull request merged**
